pages/apis/graphql/graphql_resource_limits.md

```diff
@@ -0,0 +1,181 @@
+
+Buildkite's API has a requested complexity limit of 50,000 for each individual query. This limit is enforced prior to query execution. The intention of this limit is to prevent users from requesting an excessive number of resources in a single query.
+
+As a best practice, we recommend breaking up queries into smaller, more manageable chunks and utilizing pagination to navigate through the resulting list rather than relying on a single large query.
+
+If the query exceeds the limit, the response will return HTTP 200 status code with the following error.
+
+```json
+{
+  "errors": [
+    {
+      "message": "Query has complexity of 251503, which exceeds max complexity of 50000"
+    }
+  ]
+}
+```
+
+### Time-based rate limit
+
+To ensure optimal performance, an organization can use up to 20,000 actual complexity points within a 5-minute period. By allowing a set number of actual complexity points, you have the flexibility to run queries of different sizes within a 5-minute window.
+
+As a best practice, we recommend utilizing client-side strategies like the following to manage time-based rate limits:
+
+- Caching to lower the number of API calls.
+- Queues to schedule API calls.
+- Pagination to only request the necessary data.
+
+If an organization exceeds the 20,000 point limit, the response will return HTTP 429 status code with the following error.
+
+```json
+{
+    "errors": [
+        {
+            "message": "Your organization has exceeded the limit of 20000 complexity points. Please try again in 187 seconds."
+        }
+    ]
+}
+```
+
+
+## Accessing limit details
+
+### Check time based limits
+
+The rate limit status is available in the following response headers of each GraphQL call:
+
+`RateLimit-Remaining` — The remaining complexity left within the current time window.  
+`RateLimit-Limit` — The complexity limit for the time window.  
+`RateLimit-Reset` — The number of seconds remaining until a new time window is started and the limits are reset.
+
+```js
+RateLimit-Remaining: 20
+RateLimit-Limit: 20000
+RateLimit-Reset: 120
+```
+
+### View query complexity
+
+To include the complexity data in responses, use the `Buildkite-Include-Query-Stats` header in GraphQL requests. This returns the complexity data in the response like the following:
+
+```json
+{
+  "data" : {
+    "organization": {
+      "name": "Buildkite"
+    }
+  },
+  "stats" : {
+    "requestedComplexity": 1910,
+    "actualComplexity": 550
+  }
+}
+```
+
+## Best practices to avoid rate limit errors
+
+Designing your client application with best practices in mind is the simplest way to avoid throttling errors. For example, you can stagger API requests in a queue and do other processing tasks while waiting for the next queued job to run.
+
+Consider the following best practices when designing you API usage:
+
+- Optimize the request by only requesting data you require. We recommend using specific queries rather than a single all-purpose query.
```

Comment:
Missing "the":

```suggestion
- Optimize the request by only requesting the data you require. We recommend using specific queries rather than a single all-purpose query.
```